doc/md/guides/migration-tools/golang-migrate.md

```diff
@@ -0,0 +1,218 @@
+---
+id: golang-migrate
+title: Automatic migration planning for golang-migrate
+slug: /guides/migration-tools/golang-migrate
+---
+
+## TL;DR
+* [`golang-migrate`](https://github.com/golang-migrate/migrate) is a popular database migration
+CLI tool and Go library that's widely used in the Go community.
+* [Atlas](https://atlasgo.io) is an open-source tool for inspecting, planning, verifying and
+ executing schema changes to your database. 
+* Developers using `golang-migrate` can use Atlas to automatically plan schema migrations
+  for them, based on the desired state of their schema instead of crafting them by hand. 
+
+## Automatic migration planning for golang-migrate
+
+Atlas can automatically plan database schema migrations for developers using `golang-migrate`.
+Atlas plans migrations by calculating the diff between the _current_ state of the database,
+and it's _desired_ state. 
+
+For golang-migrate users, the current state can be thought of as the sum of 
+all _up_ migrations in a migration directory. The desired state can be provided to Atlas
+via an Atlas schema [HCL file](https://atlasgo.io/atlas-schema/sql-resources) or as a
+connection string to a database that contains the desired schema.
+
+In this guide, we will show how Atlas can automatically plan schema migrations for
+golang-migrate users. 
+
+## Prerequisites
+
+* An existing project with a `golang-migrate` migrations directory.
+* Docker
+* Atlas ([installation guide](https://atlasgo.io/getting-started/#installation))
+
+## Dev database
+To plan a migration from the current to the desired state, Atlas uses a [Dev Database](/concepts/dev-database),
+which is usually provided by a locally running container with an empty database of the type
+you work with (such as MySQL or PostgreSQL). 
+
+To spin up a local MySQL database that will be used as a dev-database in our example, run:
+
+```text
+docker run --rm --name atlas-db-dev -d -p 3306:3306 -e MYSQL_DATABASE=dev -e MYSQL_ROOT_PASSWORD=pass  mysql:8
+```
+
+As reference for the next steps, the URL for the dev-database will be:
```

Comment:
```suggestion
As reference for the next steps, the URL for the Dev Database will be:
```
You call it `Dev Database` before, now it is `dev-database`. We should use the same notation every time.